status/conditions_test.go

```diff
@@ -69,198 +62,435 @@ func withLastTransitionTime(c Condition, t time.Time) Condition {
 	return c
 }
 
-func TestConditionDeepCopy(t *testing.T) {
-	a := generateCondition("A", corev1.ConditionTrue)
-	var aCopy Condition
-	a.DeepCopyInto(&aCopy)
-	if &a == &aCopy {
-		t.Errorf("Expected and actual point to the same object: %p %#v", &a, &a)
-	}
-	if &a.Status == &aCopy.Status {
-		t.Errorf("Expected and actual point to the same object: %p %#v", &a.Status, &a.Status)
-	}
-	if &a.Reason == &aCopy.Reason {
-		t.Errorf("Expected and actual point to the same object: %p %#v", &a.Reason, &a.Reason)
-	}
-	if &a.Message == &aCopy.Message {
-		t.Errorf("Expected and actual point to the same object: %p %#v", &a.Message, &a.Message)
-	}
-}
-
-func TestConditionsSetEmpty(t *testing.T) {
-	conditions := initConditions()
-
-	setCondition := generateCondition("A", corev1.ConditionTrue)
-	assert.True(t, conditions.SetCondition(setCondition))
-
-	expectedCondition := withLastTransitionTime(setCondition, initTime.Add(clockInterval))
-	actualCondition := conditions.GetCondition(setCondition.Type)
-	assert.Equal(t, 1, len(conditions))
-	assert.Equal(t, expectedCondition, *actualCondition)
-}
-
-func TestConditionsSetNotExists(t *testing.T) {
-	conditions := initConditions(generateCondition("B", corev1.ConditionTrue))
-
-	setCondition := generateCondition("A", corev1.ConditionTrue)
-	assert.True(t, conditions.SetCondition(setCondition))
-
-	expectedCondition := withLastTransitionTime(setCondition, initTime.Add(clockInterval))
-	actualCondition := conditions.GetCondition(expectedCondition.Type)
-	assert.Equal(t, 2, len(conditions))
-	assert.Equal(t, expectedCondition, *actualCondition)
-}
-
-func TestConditionsSetExistsIdentical(t *testing.T) {
-	existingCondition := generateCondition("A", corev1.ConditionTrue)
-	conditions := initConditions(existingCondition)
-
-	setCondition := existingCondition
-	assert.False(t, conditions.SetCondition(setCondition))
-
-	expectedCondition := withLastTransitionTime(setCondition, initTime)
-	actualCondition := conditions.GetCondition(expectedCondition.Type)
-	assert.Equal(t, 1, len(conditions))
-	assert.Equal(t, expectedCondition, *actualCondition)
-}
-func TestConditionsSetExistsDifferentReason(t *testing.T) {
-	existingCondition := generateCondition("A", corev1.ConditionTrue)
-	conditions := initConditions(existingCondition)
-
-	setCondition := existingCondition
-	setCondition.Reason = "ChangedReason"
-	assert.True(t, conditions.SetCondition(setCondition))
-
-	expectedCondition := withLastTransitionTime(setCondition, initTime)
-	actualCondition := conditions.GetCondition(expectedCondition.Type)
-	assert.Equal(t, 1, len(conditions))
-	assert.Equal(t, expectedCondition, *actualCondition)
-}
-
-func TestConditionsSetExistsDifferentStatus(t *testing.T) {
-	existingCondition := generateCondition("A", corev1.ConditionTrue)
-	conditions := initConditions(existingCondition)
+var _ = Describe("Testing Conditions.go\n", func() {
 
```

Comment:
Put all of the tests associated with:
- the `Condition` type in a `Describe("Condition", func() {})` section
- the `Conditions` type in a `Describe("Conditions", func() {})` section

And anytime you're testing a function associated with that type, put those tests in a nested `Describe`. e.g.:

```go
Describe("Condition", func() {
	Describe("DeepCopyInto", func() {
		var a, aCopy Condition
		BeforeEach(func() {
		})
		AfterEach(func() {
		})
		It("should copy type") {}
		It("should copy status") {}
		It("should copy reason") {}
		It("should copy message") {}
		It("should copy last transition time") {}
    })
    Describe("IsTrue", func() {
	})
    Describe("IsFalse", func() {
	})
    Describe("IsUnknown", func() {
	})
})

Describe("Conditions", func() {

})
```

The hierarchy should pretty much match the hierarchy of the generated Go docs: https://pkg.go.dev/github.com/operator-framework/operator-lib/status?tab=doc

Reply:
I would simplify the DeepCopy test a bit more. I would actually just remove the `Context`s 
```golang
var _ = Describe("Condition", func() {
    Describe("DeepCopyInto", func() {
        var aCopy Condition
	a := generateCondition("A", corev1.ConditionTrue)

        It("should copy one condition to another in its entirety", func() {
            a.DeepCopyInto(&aCopy)

            Expect(aCopy.Type).To(Equal(a.Type))
            Expect(aCopy.Status).To(Equal(a.Status))
            Expect(aCopy.Reason).To(Equal(a.Reason))
            Expect(aCopy.Message).To(Equal(a.Message))
            Expect(aCopy.LastTransitionTime).To(Equal(a.LastTransitionTime))
        }
    }
    // ...
}
```

Reply:
Then for the other methods I would elaborate on what I did above and add another Describe for each of the methods.

```golang
var _ = Describe("Condition", func() {
    Describe("DeepCopyInto", func() {
        // ...
    }
    Describe("IsTrue", func() {
        // ...
    }
    Describe("IsFalse", func() {
        // ...
    }
    Describe("IsUnknown", func() {
        // ...
    }
}
```